python/lsst/cp/pipe/bfCoeffCalc.py

```diff
@@ -0,0 +1,1272 @@
+        dtype=str,
+        doc="The key by which to pull a detector from a dataId, e.g. \'ccd\'' or \'detector\'",
+        default='ccd',
+    )
+    maxIterRegression = pexConfig.Field(
+        dtype=int,
+        doc="Maximum number of iterations for the iterative regression fitter",
+        default=10
+    )
+    nSigmaClipGainCalc = pexConfig.Field(
+        dtype=int,
+        doc="Number of sigma to clip to during gain calculation",
+        default=5
+    )
+    nSigmaClipRegression = pexConfig.Field(
+        dtype=int,
+        doc="Number of sigma to clip to during iterative regression",
+        default=3
+    )
+    xcorrCheckRejectLevel = pexConfig.Field(
+        dtype=float,
+        doc="Sanity check level for the sum of the input cross-correlations. Arrays which "
+        "sum to greater than this are discarded before the clipped mean is calculated.",
+        default=1.0  # xxx change this back once the problem is fixed!
+        # default=0.2
+    )
+    maxIterSOR = pexConfig.Field(
+        dtype=int,
+        doc="The maximum number of iterations allowed for the successive over-relaxation method",
+        default=10000
+    )
+    eLevelSOR = pexConfig.Field(
+        dtype=float,
+        doc="The target residual error for the successive over-relaxation method",
+        default=5.0e-14
+    )
+    nSigmaClipKernelGen = pexConfig.Field(
+        dtype=float,
+        doc="Number of sigma to clip to during pixel-wise clipping when generating the kernel",
+        default=4
+    )
+    nSigmaClipXCorr = pexConfig.Field(
+        dtype=float,
+        doc="Number of sigma to clip when calculating means for the cross correlation",
+        default=5
+    )
+    maxLag = pexConfig.Field(
+        dtype=int,
+        doc="The maximum lag to use when calculating the cross correlation/kernel",
+        default=8
+    )
+    nPixBorderGainCalc = pexConfig.Field(
+        dtype=int,
+        doc="The number of border pixels to exclude when calculating the gain",
+        default=10
+    )
+    nPixBorderXCorr = pexConfig.Field(
+        dtype=int,
+        doc="The number of border pixels to exclude when calculating the cross correlation/kernel",
+        default=10
+    )
+    biasCorr = pexConfig.Field(
+        dtype=float,
+        doc="A scary, empirically determined correction-factor correcting for sigma-clipping" +
+        " a non-Gaussian distribution",
+        default=0.9241
+    )
+    backgroundBinSize = pexConfig.Field(
+        dtype=int,
+        doc="Size of the background bins",
+        default=128
+    )
+    fixPtcThroughOrigin = pexConfig.Field(
+        dtype=bool,
+        doc="Contrain the fit of the PTC to go through the origin?",
+        default=True
+    )
+    level = pexConfig.ChoiceField(
+        doc="The level at which to calculate kernels",
+        dtype=str, default="CCD",
+        allowed={
+            "AMP": "Every amplifier treated separately",
+            "CCD": "One kernel per CCD",
```

Comment:
Haha, I hope this too! It was me commenting that on RFC-484, and who made those changes to `obs_lsstCam`, so this is quite amusing that I went wrong here 😛

So yes, totally agree, and only 68 instance to change in this file 🤪

Reply:
This might, however, lead to some weirdness/slight inconsistency, as the dataIds will still contain CCD as most obs_packages haven't been changed yet. I'll change it where I can, and we'll see where we end up is probably best. I'll let you know when it's time to rereview.

Reply:
OK, this has now been done. I think all the places that ccd remains are more or less appropriate. It's slightly ironic that the convention we've chose to allow for dynamic keying of this is `ccdKey` rather than `detectorKey`, but changing this here and not elsewhere is both asking for trouble and well beyond the scope I think. If there are any instances you still object to do shout though.